src/Resolver.js

```diff
@@ -29,25 +30,23 @@ export default class Resolver extends React.Component {
     resolve: React.PropTypes.object,
   }
 
-  static render = function(render, node) {
-    const initialData = window.__REACT_RESOLVER_PAYLOAD__;
-
+  static render = function(render, node, data = window[PAYLOAD]) {
```

Comment:
This was the crux of the solution: If the DOM has content but the payload is missing, _fetch the payload & call `Resolver.render` with it :)

Sometimes the best answer for recursion is recursion!